src/app/hooks/use-push-notifications.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+
+import { initializeApp } from 'firebase/app';
+import { getMessaging, getToken } from 'firebase/messaging';
+import browser from 'webextension-polyfill';
+
+import { AnalyticsEventCategory, useAnalytics } from 'lib/analytics';
+import { EnvVars } from 'lib/env';
+import { useLocalStorage } from 'lib/ui/local-storage';
+
+export const usePushNotifications = () => {
+  const [fcmToken, setFcmToken] = useLocalStorage<string | undefined>('fcmToken', undefined);
+  const { trackEvent } = useAnalytics();
+
+  useEffect(() => {
+    if ('serviceWorker' in navigator && 'Notification' in window) {
```

Comment:
Please consider using an immediate return with the inverted condition to prevent excessive nesting